spring-context/src/main/java/org/springframework/cache/support/AbstractCacheManager.java

```diff
@@ -170,7 +177,7 @@ private void updateCacheNames(String name) {
 	 * or simply the passed-in Cache object by default
 	 */
 	protected Cache decorateCache(Cache cache) {
-		return cache;
+		return this.cacheDecorator != null ? this.cacheDecorator.decorateCache(cache) : cache;
```

Comment:
Moving that bits above and allowing the user to change the decorator via a public setter means you can completely shadow what `AbstractTransactionSupportingCacheManager` used to du using public API.

Also, if `CacheDecorator` is a first-class concept, you should probably expect that more than one decorator can apply at a given time. This wrapping shouldn't be the responsibility of the caller.